Add `check_for_pole` to some poles

sonicSMASH opened the pull request:
This enables signs from the street_signs mod to use this mod's poles as a pole to correctly attach to.

sonicSMASH committed:
Add `check_for_pole` to some poles

This enables signs from the street_signs mod to use this mod's poles as a pole to correctly attach to.

sonicSMASH committed:
Update mod.conf

SmallJoker commented on streets_poles/init.lua:
```diff
@@ -11,7 +11,7 @@ minetest.register_node(":streets:bigpole", {
 	drawtype = "nodebox",
 	tiles = { "streets_pole.png" },
 	sunlight_propagates = true,
-	groups = { cracky = 1, level = 2, bigpole = 1, not_blocking_trains = 1},
```
What does `not_blocking_trains` do? Why can it be removed?

EDIT: Added in https://github.com/minetest-mods/streets/pull/109

@Emojigit What does this do?

sonicSMASH replied:
I don't know what it's there for, I didn't add it for any reason.

Emojigit replied:
This is here for advtrains. If a block is registered with this group, trains will not be blocked even if this block is in its path. This is useful as poles are not full blocks, hence not visually blocking trains but technically would.

sonicSMASH replied:
Alright, it got added back.

SmallJoker commented on streets_poles/init.lua:
```diff
@@ -39,7 +39,9 @@ minetest.register_node(":streets:bigpole", {
 		["b"] = { 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1 }
-	}
+	},
+	check_for_pole = true
```
This should be explained so that the next modder knows what the purpose is.
```suggestion
	check_for_pole = true -- allows street_signs to use this as a pole to attach signs
```

SmallJoker commented on mod.conf:
```diff
@@ -1,7 +1,7 @@
-optional_depends = creative, xpanes, stairsplus, moreblocks, digilines, concrete
+optional_depends = creative, xpanes, stairsplus, moreblocks, digilines, concrete, signs_lib
 license = MIT
 version = 2.0.1
```
Perhaps increase the version to 2.1.0 ?

sonicSMASH committed:
Update streets_poles/init.lua

Co-authored-by: SmallJoker <SmallJoker@users.noreply.github.com>

sonicSMASH committed:
Change version

I am untrained in version numbers, an explanation to why it goes from 2.0.1 to 2.1.0 would be nice

sonicSMASH committed:
Undo removing `not_blocking_trains`

SmallJoker commented:
@sonicSMASH There a few nodes that still have the `not_blocking_trains = 1` group removed. Is that intentional or did you forget to undo those changes?

SmallJoker commented:
fd02b1a
